machineconfiguration/v1alpha1/types_machineconfigstate.go

```diff
@@ -0,0 +1,125 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigState describes the health of the Machines on the system
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigState struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Spec MachineConfigStateSpec `json:"spec"`
+	// +optional
+	Status MachineConfigStateStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigStateList describes all of the MachinesStates on the system
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigStateList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineConfigState `json:"items"`
+}
+
+// MachineConfigStateSpec describes the type of State we are managing
+type MachineConfigStateSpec struct {
+	// configuration describes the object reference information of a MachineConfigState
+	Config MachineConfigStateConfig `json:"configuration"`
+}
+
+// MachineConfigStateStatus holds the reported information on a particular MachineConfigState
+type MachineConfigStateStatus struct {
+	corev1.ObjectReference `json:",inline"`
+	// conditions represents the latest available observations of current state.
+	// +listType=atomic
+	// +optional
+	Conditions []ProgressionCondition `json:"conditions"`
+	// mostRecentError is populated if the State reports an error.
+	MostRecentError string `json:"mostRecentError"`
+	// health reports the overall status of this MachineConfigState Given its Progress
+	Health MachineConfigStateHealthEnumeration `json:"health"`
+	// observedGeneration represents the generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// configVersion holds the current and desired config versions
+	ConfigVersion MachineConfigVersion `json:"configVersion"`
```

Comment:
This should be expanded, primarily to explain what is updating this and when
```suggestion
	// configVersion holds the current and desired config versions for the node targeted by this MachineConfigState resource.
	// The current version represents the current machine config for the node and is updated after a successful update.
	// The desired version ...
	ConfigVersion MachineConfigVersion `json:"configVersion"`
```